CHANGELOG.md

```diff
@@ -1,5 +1,18 @@
 ## Developer Changelog
 
+### January 11, 2022 WD-3815
+* backend
+    * added unified logger (facilities/logging.js)
+      * updated fastify to use new logger
+      * updated database to use new logger
+    * moved database auto-deploy logic to a more appropriate spot (startup, not every db check)
+    * cleanup server.js: moved shutdown handlers into daemon.js
+    * restructuring
+      * new directory: facilities: for full-on components, not just helpers
+      * moved daemon.js to facilities directory
+      * helpers/auth.js -> facilities/keycloak.js because keycloak items are what is in there, and it is more than helpers
+      * helpers/server.js -> facilities/fastify.js because fastify configuration items are in there, and fastify is not just a helper
```

Comment:
@CR-gbc Is there a particular architectural pattern you're going with? Just curious about the naming convention of "facilities"

Reply:
fixing testing and openshift is coming in another PR. in keeping with our (re)quest for smaller PRs, it is not here yet. IIRC automated testing for API is not enabled at this time, rather than half-fix testing here, i will fix it all at once in the testing PR.

Reply:
@CR-gbc.... @Baelx has a good point regarding the testing, having said that, I can also agree with @CR-gbc logic, however you should probably state this in your PR's description it is a partial pr, otherwise we can only assume it is a complete pr.

Hopefully the linter and testing will be in a github action that will fail this anyway, which needs to be implemented.
@Baelx If you are ok that this is a partial pr, then you can remove your NW.

Reply:
@ShawnTurple @CR-gbc Based on what Craig said, this isn't a partial PR for this feature branch, but tests have their own dedicated feature branch coming later. I know we've sometimes had tests as their own ticket in the past, but I think we should avoid that approach for this project. We already have our testing framework set up. All you need to do is update the tests in a couple places. Since each of these sub-tasks are relatively large, I think each of them should be held to the DoD - which would include testing. Curious to know thoughts on this.

My goal is here is just to make sure testing gets done, and preferably done alongside newly-added functions for continuity. I think the size of these PRs warrants it. And I don't think we should be leaving out parts of our DoD in the name of small PRs either but that's just my two cents.

I'll greenlight this PR but I think we should be including tests more often. Why have github actions set as merge blockers if we're just turning them off most of the time? I personally don't think we should have large tickets where tests only come at the end, but they should be added with each bit of work.

Reply:
all of the testing needs fixing for the backend right now. its in a separate ticket so that this didnt become another monolith, and i think its better to fix all of the testing in that one scope, even if it is suboptimal to do it in this manner at this time. we won't do it this way again, because the ticket in question will fix all of it and get us off on the right foot.